### ISSUE TITLE
ProhibitEmptyQuotes should be configurable

### ISSUE THREAD

**Issue by mkende** [Perl-Critic/Perl-Critic#1082]
The `ProhibitEmptyQuotes` policy has no configuration option. In the spirit of other policies, I believe that it should be configurable to specify the maximum length of string that is allowed by this policy (in particular to allow empty strings to be quoted).

If this sounds reasonable, I wrote #1077 to implement such an option.

**Comment by guest20** [comment]
Are you suggesting that `ProhibitEmptyQuotes` should be configurable to allow empty quotes? 

I think you can get that by not enabling this policy.

**Comment by petdance** [comment]
I'm not understanding what you're suggesting. What is an example of code that trips this warning that you would like to be able to configure past?

**Comment by guest20** [comment]
Looking at the patch, I think @mkende wants `""` itself to be Ok, but not longer strings that are all non-printable?

I'm betting it's for cases like ` "" . $variable ` for stringification

**Comment by mkende** [comment]
Yes, in general I agree with this policy, so I don’t want to fully disable it, except in the case of empty strings that I find quite readable with quotes. Hence #1077, that would allow to configure that behavior.

**Comment by petdance** [comment]
I'm not understanding what you're suggesting. What is an example of code that trips this warning that you would like to be able to configure past?

**Comment by mkende** [comment]
> I'm not understanding what you're suggesting. What is an example of code that trips this warning that you would like to be able to configure past?

Anything like `my $var = '';`

Looking at my code an example that comes back from time to time is:

```
local $Data::Dumper::Pad = '';  ## no critic (ProhibitEmptyQuotes)
```

**Comment by petdance** [comment]
@mkende   So you have this code

    local $Data::Dumper::Pad = '';

and even though the policy throws a warning, the code is OK with you because that's easier to read than

    local $Data::Dumper::Pad = q{};

I agree with you, and that's why I never have this policy turned on.

Why don't you turn off the policy? What would you configure? How can you configure the concept of "empty"?

**Comment by mkende** [comment]
because I still believe that the following, which would be another common pattern, is bad:

```
local $Data::Dumper::Pad = '    ';
```

And that is also caught by this policy (empty is a misnomer in the policy name, it’s not just empty strings, it’s actually any string where all the characters are white spaces).

**Comment by petdance** [comment]
So here's our situation as I see it.

* ProhibitEmptyQuotes is really ProhibitStringsThatAreNothingButWhitespace. We can't change it.
* You want a way to allow a zero-length string.
* You don't really want a configurable length. It's not like two character all-whitespace strings are OK but three is bad.

So it seems to me the config option wouldn't be a length, but would be a `AllowZeroLengthStrings`, yes?

**Comment by mkende** [comment]
It could be yes, but I can also see reasons to allow patterns like `' ' x 4` or similar. So such things would be covered by a single options.

Do you see a downside to the more generic option?

**Comment by petdance** [comment]
The downside is it's more complex, and it's not a feature anyone has asked for.

**Comment by mkende** [comment]
Well, let’s say that I’m asking for it specifically for the code in my previous comment (so just two booleans `AllowZeroLengthStrings` and `AllowOneCharacterStrings` would be enough too and I promise not to ask for the `AllowTwoCharactersStrings` after that 😄).

But my thinking was that the length is maybe a little more complex for the user but is a very simple implementation, so it fixes that issue now and in the future even if nobody asked for it yet. Which I assume is valuable given limited manpower to maintain perl-critic.

Anyway, it’s your project so I’ll let you decide which approach you prefer.
